Build warning: parameter 'nBands' shadows member inherited from type 'GDALDataset'

## Expected behavior and actual behavior.

While building, there is a warning:

```
[751/1323 13.4/sec] Building CXX object frmts/georaster/CMakeFiles/gdal_GEOR.dir/georaster_dataset.cpp.o
../frmts/georaster/georaster_dataset.cpp:821:44: warning: parameter 'nBands' shadows member inherited from type 'GDALDataset' [-Wshadow-field]
                                       int nBands,
                                           ^
../gcore/gdal_priv.h:370:17: note: declared here
    int         nBands = 0;
                ^
1 warning generated.
```

Expected behavior: build without warnings


## Steps to reproduce the problem.

Build with cmake defaults, clang10

## Operating system

Ubuntu 20.04 64 bit

## GDAL version and provenance

current master, self-built

--- a/frmts/georaster/georaster_dataset.cpp
+++ b/frmts/georaster/georaster_dataset.cpp
@@ -818,7 +818,7 @@ char** GeoRasterDataset::GetFileList()
 GDALDataset *GeoRasterDataset::Create( const char *pszFilename,
                                        int nXSize,
                                        int nYSize,
-                                       int nBands,
+                                       int nBandsIn,
                                        GDALDataType eType,
                                        char **papszOptions )
 {
@@ -865,7 +865,7 @@ GDALDataset *GeoRasterDataset::Create( const char *pszFilename,
 
     poGRW->nRasterColumns   = nXSize;
     poGRW->nRasterRows      = nYSize;
-    poGRW->nRasterBands     = nBands;
+    poGRW->nRasterBands     = nBandsIn;
     poGRW->sCellDepth       = pszCellDepth;
     poGRW->nRowBlockSize    = DEFAULT_BLOCK_ROWS;
     poGRW->nColumnBlockSize = DEFAULT_BLOCK_COLUMNS;
@@ -993,9 +993,9 @@ GDALDataset *GeoRasterDataset::Create( const char *pszFilename,
     }
     else
     {
-        if( nBands == 3 || nBands == 4 )
+        if( nBandsIn == 3 || nBandsIn == 4 )
         {
-            poGRW->nBandBlockSize = nBands;
+            poGRW->nBandBlockSize = nBandsIn;
         }
     }
 

